src/core/lib/surface/server.h

```diff
@@ -74,10 +74,47 @@
 #define GRPC_ARG_SERVER_MAX_PENDING_REQUESTS_HARD_LIMIT \
   "grpc.server.max_pending_requests_hard_limit"
 
+namespace grpc {
```

Comment:
Having code in core depend on a type in the C++ layer seems a little ugly, even if just as a forward declaration.  I suspect this will go away with my comment elsewhere about synchronization, but if not, the alternative would be to make `PassiveListener` and `PassiveListenerImpl` C-core types instead of C++ types.

Reply:
Regarding the thought process here: `PassiveListener` is in the public C++ API, and I wasn't sure how to structure it so that core does not _actually_ depend on C++ code. It'd be pretty odd to have a C++-only public API class in the core namespace, right?

Reply:
Indeed, the problem is gone.